ReactiveCocoa/Swift/Signal.swift

```diff
@@ -125,7 +125,8 @@ public final class Signal<Value, Error: ErrorType> {
 	///
 	/// Returns a Disposable which can be used to disconnect the observer. Disposing
 	/// of the Disposable will have no effect on the Signal itself.
-	public func observe(observer: Observer) -> Disposable? {
+	@discardableResult
```

Comment:
I opened a separate issue to discuss this. https://github.com/ReactiveCocoa/ReactiveCocoa/issues/3039